lib/spotlight/dor/indexer.rb

```diff
@@ -61,27 +85,52 @@ def add_donor_tags sdb, solr_doc
       insert_field solr_doc, 'donor_tags', donor_tags, :symbol # this is a _ssim field
     end
 
+    # add the folder number to solr_doc as folder_ssi field (note: single valued!)
+    #   data in location/physicalLocation or in relatedItem/location/physicalLocation
+    # TODO:  push this up to stanford-mods gem?  or should it be hierarchical series/box/folder?
+    def add_folder(sdb, solr_doc)
+      # see spec for data from actual collections
+      #   _location.physicalLocation should find top level and relatedItem
+      folder_num = sdb.smods_rec._location.physicalLocation.map do |node|
+        val = node.text
+        match_data = val.match(/Folder:? ?(.+)/i)
+        next if match_data.blank?
+        result = match_data[1].rstrip
+        # Menuez collection may have folder followed by Sleeve then Frame
+        match2_data = result.match(/(.*),? ?Sleeve/i)
```

Comment:
It would be nice if we didn't have to code for specific collections (and really nice if our physical location data had any kind of internal consistency...). 

These two cases seem to be the most problematic:

- `MSS Photo 451, Series 1, Box 32, Folder 42, Sleeve 32-11-2, Frame B32-F11-S2-6`
- `Series Biographical Photographs | Box 1 | Folder Abbot, Nathan`

I wonder if we could devise a solution that solves both, and the generic case. Maybe we could try something like tokenizing on `|`, and if that fails, on `,`? I guess we'd still run into trouble with data like `Box 1, Folder Abbot, Nathan`, but hopefully that's the exception..

Also, do we know if there's any standard practice going forward that we could prioritize?